src/core/memory.cc

```diff
@@ -504,29 +474,35 @@ class small_pool_array {
 static constexpr size_t max_small_allocation
     = small_pool::idx_to_size(small_pool_array::nr_small_pools - 1);
 
-constexpr size_t object_size_with_alloc_site(size_t size) {
+constexpr size_t object_size_with_alloc_site(size_t size, bool is_sampled) {
+    (void) is_sampled;
 #ifdef SEASTAR_HEAPPROF
     // For page-aligned sizes, allocation_site* lives in page::alloc_site, not with the object.
     static_assert(is_page_aligned(max_small_allocation), "assuming that max_small_allocation is page aligned so that we"
             " don't need to add allocation_site_ptr to objects of size close to it");
-    size_t next_page_aligned_size = next_page_aligned(size);
-    if (next_page_aligned_size - size > sizeof(allocation_site_ptr)) {
-        size += sizeof(allocation_site_ptr);
-    } else {
-        return next_page_aligned_size;
+    if (is_sampled)
+    {
```

Comment:
Please keep to the coding style which puts the brackets on the same line as the if.

Reply:
Fixed a bunch of those.